source/scripts/dataHandling.js

```
@@ -51,21 +54,45 @@ window.addEventListener("DOMContentLoaded", () => {
     const image = data.get("image");
     const imageURL = await fileToDataUrl(image);
     const date = new Date();
+    const locationTag = data.get("location");
+    const moodTags = data.getAll("mood");
     const post = {
       title: title,
       description: description,
       dateCreated: date,
       image: imageURL,
+      location: locationTag,
+      mood: moodTags,
     };
 
     addMemory(post, db).then(() => displayLatestMemory(db));
     event.target.reset();
   });
-  document.getElementById("delete-all").addEventListener("click", () => {
-    deleteAllMemories(db);
+  const imageInput = document.getElementById("imageUpload");
+  const imagePreview = document.getElementById("imagePreview");
+
+  imageInput.addEventListener("change", () => {
+    const file = imageInput.files[0];
```

might want to add a check to make sure that the files array is not greater than 1, if it is maybe return an error?